docs/edgeql/select.rst

```diff
@@ -0,0 +1,671 @@
+Below, we use a subquery to select all movies containing a villain's nemesis.
+
+.. code-block:: edgeql-repl
+
+  db> select Villain {
+  ...   name,
+  ...   nemesis_name := .nemesis.name,
+  ...   movies_with_nemesis := (
+  ...     select Movie { title }
+  ...     filter Villain.nemesis in .characters
+  ...   )
+  ... };
+  {
+    default::Villain {
+      name: 'Loki',
+      nemesis_name: 'Thor',
+      movies_with_nemesis: {
+        default::Movie {title: 'Thor'},
+        default::Movie {title: 'Thor: The Dark World'},
+        default::Movie {title: 'Thor: Ragnarok'},
+        default::Movie {title: 'The Avengers'},
+      },
+    },
+    ...
+  }
+
+.. _ref_eql_select_polymorphic:
+
+Polymorphic queries
+-------------------
+
+:index: poly polymorphism nested shapes
+
+All queries thus far have referenced concrete object types: ``Hero`` and
+``Villain``. However, both of these types extend the abstract type ``Person``,
+from which they inherit the ``name`` property.
+
+Polymorphic sets
+^^^^^^^^^^^^^^^^
+
+It's possible to directly query all ``Person`` objects; the resulting set with
+be a mix of ``Hero`` and ``Villain`` objects (and possibly other subtypes of
+``Person``, should they be declared).
+
+.. code-block:: edgeql-repl
+
+  db> select Person { name };
+  {
+    default::Villain {name: 'Abomination'},
+    default::Villain {name: 'Zemo'},
+    default::Hero {name: 'The Hulk'},
+    default::Hero {name: 'Iron Man'},
+    ...
+  }
+
+You may also encounter such "mixed sets" when querying a link that points to an
+abstract type (such as ``Movie.characters``) or a :eql:op:`union type
+<TYPEOR>`.
+
+.. code-block:: edgeql-repl
+
+  db> select Movie {
+  ...   title,
+  ...   characters: {
+  ...     name
+  ...   }
+  ... }
+  ... filter .title = "Iron Man 2"
+  {
+    default::Movie {
+      title: 'Iron Man 2',
+      characters: {
+        default::Villain {name: 'Whiplash'},
+        default::Villain {name: 'Justin Hammer'},
+        default::Hero {name: 'Iron Man'},
+        default::Hero {name: 'Black Widow'},
+      },
+    },
+  }
+
+
+Polymorphic fields
+^^^^^^^^^^^^^^^^^^
+
+We can fetch different properties *conditional* on the subtype of each object
+by prefixing property/link references with ``[is <type>]``. This is known as a
+**polymorphic query**.
+
+.. code-block:: edgeql-repl
+
+  db> select Person {
+  ...   name,
+  ...   secret_identity := [is Hero].secret_identity,
+  ...   number_of_villains := count([is Hero].villains),
+  ...   nemesis := [is Villain].nemesis {
+  ...     name
+  ...   }
+  ... };
+  {
+    default::Villain {
+      name: 'Green Goblin',
+      secret_identity: {},
+      number_of_villains: 0,
+      nemesis: default::Hero {name: 'Spider-Man'},
+    },
+    default::Hero {
+      name: 'Spider-Man',
+      secret_identity: 'Peter Parker',
+      number_of_villains: 6,
+      nemesis: {},
+    },
+    ...
+  }
+
+This syntax is intentionally similar to the type intersection operator. In
```

Comment:
It's not just similar, it's _exactly_ the same operator.  What you're looking at is the abbreviated path, i.e. `Person[is Hero].secret_identity` can be written as `[is Hero].secret_identity`.